windows_integration/src/auth/wrapper.rs

```diff
@@ -0,0 +1,154 @@
+use once_cell::sync::Lazy;
+use std::{ffi::c_void, time::SystemTime};
+use windows::{
+    core::PSTR,
+    Win32::{
+        Foundation::*,
+        Security::{Authentication::Identity::*, Credentials::STATUS_LOGON_FAILURE},
+        System::Kernel::*,
+    },
+};
+
+use super::package::{AuthError, AuthPackage, AuthInfo};
+
+static mut AUTH_PACKAGE: Lazy<AuthPackage> = Lazy::new(|| AuthPackage::new());
+
+#[tokio::main]
```

Comment:
You can wrap tokio::main around a pub async extern like this? This blows my mind.